js/src/forum/components/DiscussionPage.js

```diff
@@ -194,10 +192,12 @@ export default class DiscussionPage extends Page {
     // posts we want to display. Tell the stream to scroll down and highlight
     // the specific post that was routed to.
     this.stream = new PostStreamState(discussion, includedPosts);
-    this.stream.goToNumber(m.route.param('near') || (includedPosts[0] && includedPosts[0].number()), true);
+    this.stream.goToNumber(m.route.param('near') || (includedPosts[0] && includedPosts[0].number()), true).then(() => {
+      this.discussion = discussion;
```

Comment:
Why are we setting this.discussion later now? What does this affect/solve?

Reply:
Since the PostStream and PostStreamScrubber components are only shown in view when `this.discussion` is set, this way we set `this.discussion` only after the initial set of posts has loaded, so that there's less time for race conditions / stuff to go wrong with PostStreamScrubber animation glitches.